build-scripts/gulp/entry-html.js

```diff
@@ -10,20 +10,29 @@ import { htmlMinifierOptions, terserOptions } from "../bundle.cjs";
 import env from "../env.cjs";
 import paths from "../paths.cjs";
 
+// Regex to match iPadOS >= 15 and macOS >= 11 (app web view) because browserlist is not capable of this
+const ipadMacOSRegex =
+  /\b(iPadOS\s*(1[5-9]|[2-9]\d)(?:\.\d+)?(?:\.\d+)?)|(macOS\s*(1[1-9]|[2-9]\d)(?:\.\d+)?(?:\.\d+)?)\b/;
```

Comment:
We cannot hard code the versions because they can change with time and/or usage stats.  I'm looking into the best solution here.

Reply:
May be we should also put Safari 16 in browserlist file to have the same target in browserlist. I agree it's not perfect but I didn't find another solution 😢

Let's me know if you find something better!

Reply:
I've put in many hours researching this now, and the only solid conclusion I've got is that Apple needs to do better. 😞 

For iPad, I think https://github.com/home-assistant/iOS/pull/2918 is the best solution (and looks like @bgoncal just merged it as I'm typing so 🥳).  There's no reason we need to allow the app to spoof the user agent to pretend it's a Mac because it won't affect the frontend result at all.  So the current regex will work now for iPad at least.

For the Mac app, that's a complete mess.  I had a discussion with @bgoncal  and continued to research, and the only version provided by Apple's API is the OS version (i.e. there's no way to query what WebKit or Safari version gets used by the web view).  And it's not provided in the default user agent string either.  So the only choice is to map it ourselves.  The only question for me becomes: should the MacOS version map to the Safari version it's shipped with, or the latest version it supports?  Based on my research, I think the former is correct, but I'm not 100% sure on that.

I started looking into how to automate this similar to the way `browserslist-useragent-regexp` works, and I think I could probably throw that together without too much trouble.  It's late here now so I'll do that tomorrow.

Regarding putting more Safari versions in the `browserslist` config, I agree.  Currently, modern is served to iOS 15.6+ and Safari 17+, which doesn't make much sense from a transpilation/polyfilling perspective.  That's just reflecting that iPhones are way more popular than Macs.  We can hard code a sync for now and maybe I'll do another PR to `browserslist` for a nicer solution.

Reply:
I agree! Thank you for the research! 🙂
That was more or less my conclusion too.

Reply:
Opened https://github.com/browserslist/browserslist/issues/838.  I'll work on the change for Mac in a bit.